Community Governance (#143)
gatsby-uc/plugins: GOVERNANCE.md

> > Note: OUr model is extremely new, and not yet binding. Once merged into the `main` branch, we will trial what's outlined here and make changes based on any feedback. All changes will go through the PR review process. This note will be removed when this governance model is seen as binding.

Reviewer: ```suggestion
> Note: Our model is extremely new, and not yet binding. Once merged into the `main` branch, we will trial what's outlined here and make changes based on any feedback. All changes will go through the PR review process. This note will be removed when this governance model is seen as binding.
```

- **Being a positive member of our community** - Go above and beyond our Code of Conduct, and commit to healthy communication in pull requests, issue discussions, Discord conversations, and interactions outside of our community (eg: no Twitter bullies)

Each level unlocks new privileges and responsibilities on Discord and GitHub. Below is a summary of each contributor level:

### Level 1 - Contributor

Have you done something (big or small) to contribute to the health, success, or growth of the GUC or its plugins? Congratulations, you're officially recognized as a contributor to the project!

#### Examples of recognized contributions

- **GitHub:** Submitting a merged pull request
- **GitHub:** Filing a detailed bug report or RFC
- **GitHub:** Updating documentation!
- Helping people on GitHub, Discord, etc.
- Answering questions on Stack Overflow, Twitter, etc.
- Blogging, Vlogging, Podcasting, and Livestreaming about GUC projects
- This list is incomplete! Similar contributions are also recognized.

#### Privileges

> At this time we have no specifics for this role. as the community grows we anticipate privleges to come. If you have ideas here please let us know!

Reviewer: ```suggestion
At this time we have no specifics for this role. As the community grows we anticipate privileges to come. If you have ideas here please let us know!
```

#### Responsibilities

This role does not require any extra responsibilities or time commitment. We hope you stick around and keep participating!

If you're interested in reaching the next level and becoming a Maintainer, you can begin to explore some of those responsibilities in the next section.

#### Nomination Process

n/a

> ### Level 2 (L2) - Maintainer

Reviewer: Should Level 1 be (L1) then, too? (In the heading)

> The **Maintainer** role is available to any contributor who wants to join the team and take part in the long-term maintenance of a plugin(s) in the GUC.

Reviewer: ```suggestion
The **Maintainer** role is available to any contributor who wants to join the team and take part in the long-term maintenance of plugins in the GUC.
```

From my standpoint it's also clear this way that it doesn't necessarily mean you maintain all plugins

> The Maintainer role is critical to the long-term health of GUC plugins. Maintainers act as the first line of defense when it comes to new issues, pull requests and Discord activity. Maintainers are most likely the first people that a user will interact with on Discord or GitHub.

Reviewer 2: ```suggestion
The Maintainer role is critical to the long-term health of GUC plugins. Maintainers act as the first line of support when it comes to new issues, pull requests and Discord activity. Maintainers are most likely the first people that a user will interact with on Discord or GitHub.
```

Author: I can understand not using the word defense. But given the Readme where we explicitly say we're not providing support this seems like a contradiction. Thoughts?

**A Maintainer is not required to write code!** Some Maintainers spend most of their time inside of Discord, maintaining a healthy community there. Maintainers can also be thought of as **Moderators** on Discord and carry special privileges for moderation.

There is no strict minimum number of contributions needed to reach this level, as long as you can show **sustained** involvement over some amount of time (at least a couple of weeks).

#### Recognized Contributions

- **GitHub:** Contributing a new or existing Gatsby plugin
- **GitHub:** Submitting non-trivial pull requests and RFCs
- **GitHub:** Reviewing non-trivial pull requests and RFCs
- **Discord:** Supporting users in Discord
- **Discord:** Active participation in calls and other events
- **GitHub + Discord:** Triaging and confirming user issues
- This list is incomplete! Similar contributions are also recognized.

#### Privileges

- All privileges of the [Contributor role](#level-1---contributor), plus...
- `@maintainer` role on [Discord](https://discord.gg/gwnqfzetjv)
- New name color on Discord: **blue**.

> - Invitation to the private #maintainers channel on Discord.

Reviewer: ```suggestion
- Invitation to the private `#maintainers` channel on Discord.
```

@pragmaticpat we want us (Gatsby) commit ourselves to also be in that channel and help out as needed. As in that sentence would also say: "[...] where also the Gatsby team can be reached"

Reviewer 2: @LekoArts - I don't see which sentence you're specifically referencing. That said, I agree that there can be a mention of a Gatsby team member that at least checks in on some frequency. We should probably consider some sense of "office hours" for supporting any questions that arise. 

How's that sound to you and @moonmeister ?

Author: Doesn't sound like that needs to be in the governance docs. If we want to mention this as a resource in the contributing docs that seems fine. Unless we want to give maintainers explicit access to the Gatsby team.

Reviewer: > If we want to mention this as a resource in the contributing docs that seems fine.

👍

- Invitation to the `maintainers` team on GitHub.
- Ability to moderate Discord.
- Ability to push branches to the repo (No more personal fork needed).
- Ability to review GitHub PRs.

> - Ability to merge _some_ GitHub PRs.

Reviewer: What's "some"?

Author: This is leftover from astro. I thought I'd leave it. ATM they could merge anything.

- Define project direction and planning

#### Responsibilities

> - Participate in the project as a team player.

Reviewer 2: ```suggestion
- Participate in the project as a team player
```

Reviewer 2: The intention here is just for consistency - either a trailing `.` or no trailing `.` on list items

In some rare cases, this role may be revoked by a project Steward. However, under normal circumstances this role is granted for as long as the contributor wishes to engage with the project.

#### Nomination Process

- You can be considered by any existing Maintainer (L2 or above).

### Steward

Steward is an additional privilege bestowed to 1 (or more) Maintainers. The role of Steward is mainly an administrative one. Stewards control and maintain sensitive project assets.

In extremely rare cases, a Steward can act unilaterally when they believe it is in the project's best interest and can prove that the issue cannot be resolved through normal governance procedure. The steward must publicly state their reason for unilateral action before taking it.

The project Steward is currently: **@moonmeister**

#### Responsibilities

- Access to the [@gatsby_uc Twitter account](https://twitter.com/gatsby_uc)
- Administration privileges on the [GUC GitHub org](https://github.com/gatsby_uc)
- Administration privileges on the [GUC Discord server](https://discord.gg/GwNqFzeTJv)
- Publish access to all GUC npm packages

> - Define project direction and planning

Reviewer: I'd say that L2 members can weigh in on that, too?

> - Ability to decide on moderation decisions

Reviewer 2: ```suggestion
- Final say on moderation decisions
```
